src/components/badge/beta_badge/beta_badge.tsx

```diff
@@ -72,67 +125,145 @@ type BadgeProps = {
    * otherwise the label will be used
    */
   title?: string;
+  /**
+   * Accepts accent, subdued and hollow.
+   */
+  color?: BetaBadgeColor;
+  size?: BetaBadgeSize;
 } & ExclusiveUnion<LabelAsNode, LabelAsString>;
 
 export type EuiBetaBadgeProps = CommonProps &
-  Omit<HTMLAttributes<HTMLSpanElement>, 'title'> &
+  ExclusiveUnion<
+    ExclusiveUnion<WithButtonProps, WithAnchorProps>,
+    WithSpanProps
+  > &
   BadgeProps;
 
 export const EuiBetaBadge: FunctionComponent<EuiBetaBadgeProps> = ({
   className,
   label,
+  color = 'hollow',
   tooltipContent,
   tooltipPosition = 'top',
   title,
   iconType,
+  onClick,
+  onClickAriaLabel,
+  href,
+  rel,
+  target,
+  size = 'm',
   ...rest
 }) => {
+  let singleLetter = false;
+  if (typeof label === 'string' && label.length === 1) {
+    singleLetter = true;
+  }
+
   const classes = classNames(
     'euiBetaBadge',
     {
       'euiBetaBadge--iconOnly': iconType,
     },
+    {
+      'euiBetaBadge--singleLetter': singleLetter,
+    },
+    {
+      'euiBetaBadge-isClickable': onClick || href,
+    },
```

Comment:
Quick suggestion 😉 . All of these can actually be combined in one object

```suggestion
    {
      'euiBetaBadge--iconOnly': iconType,
      'euiBetaBadge--singleLetter': singleLetter,
      'euiBetaBadge-isClickable': onClick || href,
    },
```